benchmark/algorithms/diskann-t2.py

```diff
@@ -0,0 +1,152 @@
+
+    def fit(self, dataset):
+        """
+        Build the index for the data points given in dataset name.
+        """
+
+        ds = DATASETS[dataset]()
+        d = ds.d
+
+        buildthreads = self._index_params.get("buildthreads", -1)
+        if buildthreads == -1:
+            buildthreads = diskannpy.omp_get_max_threads()
+
+        print("Set build-time number of threads:", buildthreads)
+        diskannpy.omp_set_num_threads(buildthreads)
+
+        metric_type = (
+                diskannpy.L2 if ds.distance() == "euclidean" else
+                1/0
+        )
+
+        index_dir = self.create_index_dir(ds)
+        self.index_path = os.path.join(index_dir, self.index_name())
+        
+        if not hasattr(self, 'index'):
+            self.index = diskannpy.DiskANNFloatIndex()
+
+        start = time.time()
+        self.index.build(ds.get_dataset_fn(), self.index_path, self.R, self.L, self.B, self.M, buildthreads)
+        end = time.time()
+        print("DiskANN index built in %.3f s" % (end - start))
+
+
+
+    def load_index(self, dataset):
+        """
+        Load the index for dataset. Returns False if index
+        is not available, True otherwise.
+
+        Checking the index usually involves the dataset name
+        and the index build paramters passed during construction.
+        """
+        ds = DATASETS[dataset]()
+        index_dir = self.create_index_dir(ds)
+        index_path = os.path.join(index_dir, self.index_name())
+
+        self.index = diskannpy.DiskANNFloatIndex()
+        if (self.index.load_index(index_path, diskannpy.omp_get_max_threads()) == 0):
+            print ("Load index success.")
+            return True
+        else:
+            return False
+
+    def query(self, X, k):
+        """Carry out a batch query for k-NN of query set X."""
+        nq, dim = (np.shape(X))
+        Ls = self._query_args.get("Ls")
+        BW = self._query_args.get("BW")
+        threads = self._query_args.get("T")
+        
+        self.res, self.query_dists = self.index.batch_search_numpy_input(X, dim, nq, k, Ls, BW, threads)
+
+    def range_query(self, X, radius):
+        """
+        Carry out a batch query for range search with
+        radius.
+        """
+        pass
+
+    def get_results(self):
+        """
+        Helper method to convert query results of k-NN search.
+        If there are nq queries, returns a (nq, k) array of integers
+        representing the indices of the k-NN for each query.
+        """
+        return self.res
+
+    def get_range_results(self):
+        """
+        Helper method to convert query results of range search.
+        If there are nq queries, returns a triple lims, I, D.
+        lims is a (nq) array, such that
+
+            I[lims[q]:lims[q + 1]] in int
+
+        are the indiices of the indices of the range results of query q, and
+
+            D[lims[q]:lims[q + 1]] in float
+
+        are the distances.
+        """
+        return self.res

```

Comment:
I would probably let this out here since it is unlikely to change from the base implementation?